hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java

```diff
@@ -93,7 +92,7 @@ public static HoodieWriteConfig createMetadataWriteConfig(
             .withCleanerParallelism(parallelism)
             .withCleanerPolicy(HoodieCleaningPolicy.KEEP_LATEST_COMMITS)
             .withFailedWritesCleaningPolicy(failedWritesCleaningPolicy)
-            .retainCommits(DEFAULT_METADATA_CLEANER_COMMITS_RETAINED)
+            .retainCommits(writeConfig.getCleanerCommitsRetained())
```

Comment:
Nice catch, can we write a test case for it?

Reply:
I've adjust the archiving config in testMetadataTableArchival() to hit the corner case, if use the MDT default clean config: DEFAULT_METADATA_CLEANER_COMMITS_RETAINED, the test will fail,  align the MDT clean config to DT makes the test pass

Reply:
Can we add a new test instead of modifying the existing test.

Reply:
A new test has been added now.